File: src/main/java/com/kerimsenturk/labreport/dto/request/UpdateUserRequest.java
package com.kerimsenturk.labreport.dto.request;

import jakarta.validation.constraints.Pattern;

import java.util.Optional;

public record UpdateUserRequest(
        //User might be change only one field or few, not whole fields.
        //That is why these fields are optional
        Optional<
                @Pattern(
                        regexp = "^([0-9]+){7,11}$",
                        message = "{pattern.unmatched.userId}")String> username,
        Optional<String> name,
        Optional<String> surname,
        Optional<
                @Pattern(
                        regexp = "^(?=.*?[a-z]+)(?=.*?[A-Z]+)(?=.*?[0-9]+).{8,20}$",
                        message = "{pattern.unmatched.password}")String> password) {
}

File: src/main/web-ui/src/domain/payload/request/UpdateUserRequest.ts
export class UpdateUserRequest {
    name?: string;
    surname?: string;
    password?: string;


    constructor(
        name?: string,
        surname?: string,
        password?: string
    ) {
        this.name = name;
        this.surname = surname;
        this.password = password;
    }
}
